### ISSUE TITLE
Dependency parsing error: verb has no subject

### ISSUE THREAD

**Issue by bongbang** [explosion/spaCy#3418]
I need to be able to identify the subject of a verb in my application, especially when the verb is the root. SpaCy is normally so good at that this error came as a nasty surprise.

How should I deal with it? I thought of using a back-up parsing logic outside spaCy to correct this, but then if I could do that reliably, then I should be able to train spaCy's model to parse better from the get-go. Would you recommend that? If so, should I update an existing model or start from scratchy? And what kind of samples and how many do I need to correct a problem like this? (I haven't read up on training documentation, but I'm going to do so right after this.) 

## How to reproduce the behaviour
In the following sentence, "Miller" is the subject (`nsubjpass`) of the verb "offered", which is the root. However, the parser somehow takes it to be a conjunct of "Irvington," leaving with verb with no subject whatsoever. (Here's a [live visualization](https://explosion.ai/demos/displacy?text=While%20visiting%20his%20hometown%20of%20Irvington%2C%20NJ%2C%20Miller%20was%20offered%20a%20job%20at%20his%20uncle%27s%20factory.&model=en_core_web_sm&cpu=1&cph=1), which made me realize that the `md` and `lg` models actually get it right. I'm letting my question stand, however, I would like to know what to do when those larger models fail similarly.)

```
def get_token_info(text, i=9):
    doc = nlp(text)
    token = doc[i]
    print(token, token.dep_, token.head)

get_token_info("While visiting his hometown of Irvington, NJ, Miller was offered a job at his uncle's factory.")
# Miller conj Irvington
```

The strange thing is, if the sentence were ever so slightly different, then spaCy would get it right, as illustrated below. In the first three cases, the difference can be just a matter of a different proper noun.

```
get_token_info("While visiting his hometown of Buffalo, NJ, Miller was offered a job at his uncle's factory.") # change city
get_token_info("While visiting his hometown of Irvington, NY, Miller was offered a job at his uncle's factory.") # change state
get_token_info("While visiting his hometown of Irvington, NJ, John was offered a job at his uncle's factory.") # change name
get_token_info("While visiting his hometown of Irvington, NY, Miller ran into a friend at his uncle's factory.") # change passive to active
get_token_info("While visiting Irvington, NJ, Miller was offered a job at his uncle's factory.", 6) # shorten adverbial clause
```

## Your Environment
* Operating System:
* Python Version Used: 3.6
* spaCy Version Used: 2.0.18
* Environment Information:

**Comment by ines** [comment]
The thing is, model's are statistical and even if you can achieve a 90% accuracy on your data (which would be considered very good), it'd still mean that every 10th prediction it makes will be wrong.

> Would you recommend that? If so, should I update an existing model or start from scratchy? And what kind of samples and how many do I need to correct a problem like this?

You most likely wouldn't want to train a new model from scratch – especially considering the pre-trained models work well for your use case, they just make mistakes because they'll never be 100% accurate. For refrence, spaCy's English models were trained on a corpus of ~2 million words, labelled with part-of-speech tags, dependencies and named entities. Even if you did the same and more, there's still no guarantee that the model would magically perform significantly better on your examples.

If you can make out a pattern of examples and constructions the models often fail on, this could indicate that examples like that were less frequent in the training data. For cases like that, you could try to update the model with more examples of those – however, you also need to make sure it doesn't overfit on the new examples and include enough examples of things that the model previously got right.

You might find that the best solution is actually to add a set of rules on top that either corrects certain mistakes, or makes it easier for the parser to correctly predict the dependency labels. For example, it seems like one of the trickier parts in your examples that the model struggles with is stuff like `"Irvington, NJ"`? Maybe you'll make it easier for the parser if you merged those into single tokens before applying the parser? There are many similar strategies to explore and ultimately, you'd have to try what works best. But I'd say it's definitely a more promising approach than trying to hack towards 100% accuracy, which is just not very realistic.
